src/control/cmd/agent/infocache.go

```diff
@@ -50,6 +51,8 @@ type attachInfoCache struct {
 	initialized atm.Bool
 	// maps NUMA affinity and device index to a response
 	numaDeviceMarshResp map[int]map[int][]byte
+	// response given when there is no numa or device data
+	defaultResp []byte
```

Comment:
I'm not super crazy about this... Wouldn't it be better to look at the length of the marshaled response map after the device scan loop and if it's empty, just create an entry for numa node 0, device 0? It's kind of a kludge either way, but this way adds more code that has to be scrutinized/maintained.

Reply:
I can do that.  I'll submit the change in https://github.com/daos-stack/daos/pull/2592